packages/message/src/repositories/MessageRepository.ts

```diff
@@ -0,0 +1,52 @@
+// import { StorageFactory } from '@adorsys-gis/storage';
+import { StorageFactory } from '@adorsys-gis/storage';
+import { DBSchema } from 'idb';
+import { Message } from '../model/Message';
+
+interface MyDatabase extends DBSchema {
+  messages: {
+    key: number;
+    value: Message;
+  };
+}
+
+export class MessageRepository {
+  private storage: StorageFactory<MyDatabase>;
+
+  constructor() {
+    this.storage = new StorageFactory<MyDatabase>('MessageDB', 1, {
+      upgrade: (db) => {
+        if (!db.objectStoreNames.contains('messages')) {
+          db.createObjectStore('messages', {
+            keyPath: 'id',
+          });
+        }
+      },
+    });
+  }
+
+  async create(message: Message): Promise<Message> {
+    await this.storage.insert('messages', { value: message });
+    return message;
+  }
+
+  async getAllByContact(contactId: string): Promise<Message[]> {
+    const allMessages = await this.storage.findAll('messages');
+    return allMessages
+      .map((item) => item.value)
+      .filter((msg) => msg.contactId === contactId);
+  }
```

Comment:
For better efficiency, I would advise to try defining an index on the contact ID field and use a `findManyByIndex` instead of `findAll` with post-filtering.